pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -27,10 +23,10 @@ const (
 	CertificateIssuer = "auth.openshift.io/certificate-issuer"
 	// CertificateHostnames contains the hostnames used by a signer.
 	CertificateHostnames = "auth.openshift.io/certificate-hostnames"
+	// RunOnceContextKey is a context value key that can be used to call the controller Sync() and make it only run the syncWorker once and report error.
+	RunOnceContextKey = "cert-rotation-controller.openshift.io/run-once"
```

Comment:
this is odd. Have never seen this.

Reply:
it is legit and I don't see other way to achieve prior behavior and keeping the `factory.Controller` interface there... I think we want to suppress snowflakes in controllers. This will give reasonable replacement for controller.RunOnce() method :-)

Reply:
am still lost when reading "context key". What is a context key?

Reply:
https://golang.org/pkg/context/#example_WithValue

Reply:
@sttts in other words... if you have run-once context passed to Sync(), return right after the syncWorker() is called with error it returns and don't try to update status or something.

Reply:
It's for tests?

Reply:
@sttts no, it is for one single snowflake controller that we need to expose this mode for (RunOnce)... I would dwell on it too much :-)

Reply:
ok, got it